ast.cpp

```diff
@@ -128,11 +178,13 @@ NodeProgram::NodeProgram(vector<NodeStmt*> _stmts){
     }   
 }
 
-NodeId::NodeId(string _id){
+NodeId::NodeId(string _id) {
     id = _id;
     if(var_map.find(id) == var_map.end()){
```

Comment:
Add a comment here noting that NodeId's c_type will be set by its parent NodeAssign because right now it needs info from the RHS. Later when we add type declarations, this should no longer be needed, since this is technically not supported in C (dynamically inferring the type of lhs by looking at rhs).